detectors/TransLab/network_runner.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import print_function
+
+import os
+import sys
+import cv2
+import logging
+import torch
+import torch.nn as nn
+import torch.utils.data as data
+
+from pathlib import Path
+from common.network_runner_base import NetworkRunnerBase
+from segmentron.utils import options
+from torchvision import transforms
+from typing import Tuple, Any
+
+from segmentron.data.dataloader import get_segmentation_dataset
+from segmentron.models.model_zoo import get_segmentation_model
+from segmentron.utils.distributed import (
+    make_data_sampler,
+    make_batch_data_sampler,
+)
+from segmentron.config import cfg
+from segmentron.utils.default_setup import default_setup
+
+
+class NetworkRunner(NetworkRunnerBase):
+    def __init__(
+        self,
+        input_dir: Path,
+        output_dir: Path,
+        log_path: Path,
+        model_path: Path,
+        segmentron_args,
+    ):
+        root_path = os.path.abspath(os.path.dirname(__file__))
+        sys.path.append(root_path)
+
+        cfg.TEST.TEST_MODEL_PATH = str(model_path)
+        cfg.DEMO_DIR = str(input_dir)
+        cfg.update_from_file("configs/trans10K/translab.yaml")
+        cfg.PHASE = "test"
+        cfg.ROOT_PATH = root_path
+        cfg.DATASET.NAME = "trans10k_extra"
+        cfg.check_and_freeze()
+
+        sys.argv = [__file__] + segmentron_args
+        args = options.parse_args()
+        default_setup(args)
+
+        self.args = args
+        self.device = torch.device(args.device)
+
+        super().__init__(input_dir, output_dir, log_path, model_path)
+
+        self._init_dataset()
+
+    def _image_gen(self) -> Tuple[Any, Path, Tuple[int, int]]:
```

Comment:
`Tuple[Any, Path, Tuple[int, int]]` looks scary. Maybe add a class that encapsulates this information